File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/publisher/acknowledge/ProductClient.java
package io.micronaut.rabbitmq.docs.publisher.acknowledge;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.rabbitmq.annotation.Binding;
import io.micronaut.rabbitmq.annotation.RabbitClient;
import io.reactivex.Completable;
import io.reactivex.Maybe;
import org.reactivestreams.Publisher;
import reactor.core.publisher.Mono;
// end::imports[]

@Requires(property = "spec.name", value = "PublisherAcknowledgeSpec")
// tag::clazz[]
@RabbitClient
public interface ProductClient {

    @Binding("product")
    Completable send(byte[] data); // <1>

    @Binding("product")
    Maybe<Void> sendMaybe(byte[] data); // <2>

    @Binding("product")
    Mono<Void> sendMono(byte[] data); // <3>

    @Binding("product")
    Publisher<Void> sendPublisher(byte[] data); // <4>
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/consumer/custom/annotation/ProductClient.py
from abc import ABC, abstractmethod

from micronaut.context.annotation import Requires
from micronaut.rabbitmq.annotation import Binding, RabbitClient


@Requires(property="spec.name", value="DeliveryTagSpec")
@RabbitClient
class ProductClient(ABC):

    @Binding("product")
    @abstractmethod
    def send(self, data: bytes) -> None:
        ...
